src/widgets/numeric-refinement-list/numeric-refinement-list.js

```diff
@@ -93,15 +94,18 @@ function numericRefinementList({
         templates
       });
 
-      this._toggleRefinement = facetValue => helper
-        .setState(refine(helper.state, attributeName, options, facetValue))
-        .search();
+      this._toggleRefinement = (facetValue) => {
+        let refinedState = refine(helper.state, attributeName, options, facetValue);
+        helper.setState(refinedState).search();
+      };
     },
     render: function({results, state, createURL}) {
-      let facetValues = options.map(facetValue => {
-        facetValue.isRefined = isRefined(state, attributeName, facetValue);
-        facetValue.attributeName = attributeName;
-        return facetValue;
```

Comment:
We still need the initial keys from the `facetValue`, so maybe something like this instead:

```
let facetValues = options.map(facetValue => {
  return {
    ...facetValue,
    isRefined: isRefined(state, attributeName, facetValue),
    attributeName: attributeName
  };
})
```

Yes, this could work. This would avoid including the `cloneDeep`.